### PR TITLE
Fix nested function array allocation

### PR DESCRIPTION
**Pull request by leirocks**

### CODE REVIEW

**Comment by leirocks** [comment]
@curtisman can you review please

File: lib/Runtime/Base/FunctionBody.cpp [hunk 1]
```diff
@@ -1143,7 +1143,7 @@ namespace Js
         if (nestedCount > 0)
         {
             nestedArray = RecyclerNewPlusZ(m_scriptContext->GetRecycler(),
-                nestedCount*sizeof(FunctionProxy), NestedArray, nestedCount);
+                (nestedCount-1)*sizeof(FunctionProxy*), NestedArray, nestedCount);
```
Comment (curtisman) [inline lib/Runtime/Base/FunctionBody.cpp]: Is it possible for NestedArray to have 0 length array to avoid this -1 here?

Comment (leirocks) [inline lib/Runtime/Base/FunctionBody.cpp]: C++ don't allow that? No?

Comment (curtisman) [inline lib/Runtime/Base/FunctionBody.cpp]: Have you tried? 
There are other example in the code base that have declares a 0 length array field.
e.g. TypePath's "assignment" field.

Comment (leirocks) [inline lib/Runtime/Base/FunctionBody.cpp]: looks c++ 11 does not allow it, but the compilers do allow zero length array in structure defination. updated the PR

**Comment by curtisman** [comment]
LGTM

**Pull request merged**